pkg/utils/kubernetes/networking/services.go

```diff
@@ -45,3 +48,92 @@ func WaitForServiceLoadBalancerAddress(ctx context.Context, c *kubernetes.Client
 		}
 	}
 }
+
+// WaitForServicePortConnected waits for a service provided by namespace/name
+// being able to connect by TCP. It creates a LoadBalancer with same selector and ports,
+// and try to establish TCP connection to the loadbalacer IP and port.
+func WaitForServicePortConnected(ctx context.Context, c *kubernetes.Clientset, namespace, name string, port int) error {
```

Comment:
What do you think about this alternative name suggestion?

```suggestion
func WaitForConnectionOnServicePort(ctx context.Context, c *kubernetes.Clientset, namespace, name string, port int) error {
```

Reply:
updated.